File: ios/Runner/Person.m
//
//  Person.m
//  Runner
//
//  Created by Yoshihiro Tanaka on 2016/12/26.
//  Copyright © 2016年 The Chromium Authors. All rights reserved.
//

#import "Person.h"

@implementation Person

- (NSDictionary *) toDictionary {
    NSMutableDictionary *dict = [[NSMutableDictionary alloc] init];
    dict[@"firstName"] = self.firstName;
    dict[@"lastName"] = self.lastName;
    dict[@"age"] = [NSNumber numberWithInt:self.age];
    dict[@"country"] = self.country;
    return dict;
}

@end

File: lib/person.dart
class Person {
  String firstName;

  String lastName;

  int age;

  String country;

  Person({this.firstName, this.lastName, this.age, this.country});

  Person.fromMap(Map map) {
    firstName = map['firstName'];
    lastName = map['lastName'];
    age = map['age'];
    country = map['country'];
  }

  Map toMap() {
    return {
      'firstName': firstName,
      'lastName': lastName,
      'age': age,
      'country': country
    };
  }

  bool validate() {
    return firstName != null &&
        lastName != null &&
        age != -1 &&
        country != null;
  }
}
